<launch>  
    <!-- 加载机器人模型到参数空间 -->
        <!-- robot_description + joint_limit + 运动学求解器 -->
        <include file="$(find robot_moveit_config4)/launch/planning_context.launch">
            <arg name="load_robot_description" value="true" />
        </include>
        <!-- 坐标变换 -->
        <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
            <rosparam param="source_list">[move_group/fake_controller_joint_states]</rosparam>
        </node>
        <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen" />
        
    <!-- 打开moveit的action server,包括环境设置server -->
        <include file="$(find robot_moveit_config4)/launch/move_group.launch">
            <arg name="allow_trajectory_execution" value="true"/>
            <arg name="moveit_controller_manager" value="fake" />
            <arg name="fake_execution_type" value="interpolate"/>
            <arg name="info" value="true"/>
            <arg name="debug" value="false"/>
            <arg name="pipeline" value="ompl"/>
            <arg name="load_robot_description" value="false"/>
        </include>
        <node name="path_plan" pkg="robot_control" type="path_plan.py" respawn="false" output="screen"/>

    <!-- 启动rviz -->
            <node pkg="rviz" type="rviz" name="rviz" args="-d $(find robot_control)/rviz/path_plan.rviz"/>
</launch>